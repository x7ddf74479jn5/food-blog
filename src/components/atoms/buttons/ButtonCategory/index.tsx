import NextLink from "@/components/atoms/NextLink";
import type { TCategory } from "@/types";
import { classNames } from "@/utils/css";
import { urlTable } from "@/utils/paths/url";

type ButtonCategoryProps = {
  category: TCategory;
};

export const ButtonCategory: React.FC<ButtonCategoryProps> = ({ category }) => {
  return (
    <NextLink href={`${urlTable.categories}/${category.slug}`}>
      <div
        className={classNames(
          "inline-block rounded-lg border-2 py-1 px-2 text-black dark:text-white",
          category.color === "blue"
            ? "bg-blue-700 dark:border-blue-300"
            : category.color === "green"
            ? "bg-green-700 dark:border-green-300"
            : category.color === "indigo"
            ? "bg-indigo-700 dark:border-indigo-300"
            : category.color === "pink"
            ? "bg-pink-700 dark:border-pink-300"
            : category.color === "purple"
            ? "bg-purple-700 dark:border-purple-300"
            : category.color === "red"
            ? "bg-red-700 dark:border-red-300"
            : category.color === "yellow"
            ? "bg-yellow-700 dark:border-yellow-300"
            : "bg-gray-700 dark:border-gray-300"
        )}
      >
        {category.name}
      </div>
    </NextLink>
  );
};
