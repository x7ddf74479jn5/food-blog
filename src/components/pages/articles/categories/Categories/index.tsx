import { TwoColumnLayout } from "@/components/layouts";
import { CategoryList } from "@/components/molecules/category/CategoryList";
import type { TCategory, TConfig, TPickup, TRankedArticle, TTag } from "@/types";
import { formatPageTitle, formatPageUrl } from "@/utils/formatter";
import { getBackLinks, urlTable } from "@/utils/paths/url";

export type CategoriesProps = {
  config: TConfig;
  categories: TCategory[];
  tags: TTag[];
  pickup: TPickup;
  popularArticles: TRankedArticle[];
};

export const Categories: React.FC<CategoriesProps> = ({ categories, config, pickup, popularArticles, tags }) => {
  const { host, siteTitle } = config;
  const heading = "カテゴリー一覧";
  const pageTitle = formatPageTitle(heading, siteTitle);
  const url = formatPageUrl(urlTable.categories, host);
  const backLinks = getBackLinks([urlTable.home]);
  return (
    <TwoColumnLayout
      config={config}
      pickup={pickup}
      host={host}
      url={url}
      title={pageTitle}
      backLinks={backLinks}
      heading={heading}
      categories={categories}
      tags={tags}
      popularArticles={popularArticles}
    >
      <CategoryList categories={categories} width={128} height={128} />
    </TwoColumnLayout>
  );
};
