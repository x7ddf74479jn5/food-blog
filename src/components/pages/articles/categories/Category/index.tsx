import { HtmlHeadBase } from "@/components/functions/meta";
import { DefaultLayout } from "@/components/layouts";
import { ArticleSWRContainer } from "@/components/organisms/article";
import type { TArticleListResponse, TCategory, TConfig, TPickup, TRankedArticle, TTag } from "@/types";
import { formatPageTitle } from "@/utils/formatter";
import { getBackLinks, urlTable } from "@/utils/paths/url";

export type CategoryProps = {
  category: TCategory;
  categories: TCategory[];
  config: TConfig;
  data: TArticleListResponse;
  pickup: TPickup;
  popularArticles: TRankedArticle[];
  tags: TTag[];
};

export const Category: React.FC<CategoryProps> = ({
  data,
  category,
  config,
  categories,
  pickup,
  tags,
  popularArticles,
}) => {
  const { siteTitle, host } = config;
  const heading = `カテゴリー：${category.name}`;
  const pageTitle = formatPageTitle(heading, siteTitle);
  const url = new URL(`${urlTable.categories}/${category.slug}`, host).toString();
  const backLinks = getBackLinks([urlTable.home, urlTable.categories]);
  const queryOptions = { filters: `category[equals]${category.id}` };

  return (
    <DefaultLayout
      config={config}
      pageTitle={pageTitle}
      url={url}
      tags={tags}
      backLinks={backLinks}
      categories={categories}
      pickup={pickup}
      popularArticles={popularArticles}
    >
      <HtmlHeadBase indexUrl={host} pageTitle={pageTitle} url={url} image={category.image.url} />
      <div className="mb-8">
        <h1>{heading}</h1>
      </div>
      <div className="min-h-screen w-full">
        <ArticleSWRContainer fallbackData={data} queryOptions={queryOptions} />
      </div>
    </DefaultLayout>
  );
};
