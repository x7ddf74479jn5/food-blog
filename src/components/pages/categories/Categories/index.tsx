import BackLinks from "@/components/feature/BackLinks";
import { ShareButtons } from "@/components/feature/ShareButtons";
import { PickupArticles, PopularArticles } from "@/components/model/article";
import { CategoryList } from "@/components/model/category/CategoryList";
import {
  AsideContainer,
  BottomAreaContainer,
  ContainerWithOrder,
  MiddleAreaContainer,
} from "@/components/ui/containers";
import { fetchConfig } from "@/repositories";
import { formatPageTitle, formatPageUrl } from "@/utils/formatter";
import { getBackLinks, urlTable } from "@/utils/paths/url";

export const Categories = async () => {
  const { host, siteTitle } = await fetchConfig();
  const pageTitle = formatPageTitle("カテゴリー一覧", siteTitle);
  const url = formatPageUrl(urlTable.categories, host);
  const backLinks = getBackLinks([urlTable.home]);

  return (
    <>
      <h1 className="my-8">カテゴリー一覧</h1>
      <MiddleAreaContainer>
        <AsideContainer className="order-2 md:order-1 lg:w-full">
          <ShareButtons url={url} title={pageTitle} className="flex-row md:flex-col" />
        </AsideContainer>
        <main className="order-1 md:order-2">
          {/* @ts-expect-error server component */}
          <CategoryList width={128} height={128} />
        </main>
      </MiddleAreaContainer>
      <BottomAreaContainer>
        <ContainerWithOrder order="order-1 lg:order-2" className="grow">
          {/* @ts-expect-error server component */}
          <PickupArticles />
          {/* @ts-expect-error server component */}
          <PopularArticles />
        </ContainerWithOrder>
        <ContainerWithOrder order="order-2 lg:order-1" className="shrink">
          <BackLinks links={backLinks} />
        </ContainerWithOrder>
      </BottomAreaContainer>
    </>
  );
};
