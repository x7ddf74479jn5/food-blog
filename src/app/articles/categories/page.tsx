import type { GetStaticProps, NextPage } from "next";

import ErrorPage from "@/app/_error/page";
import type { CategoriesProps } from "@/components/pages/articles/categories/Categories";
import { Categories } from "@/components/pages/articles/categories/Categories";
import { sentryLogServer } from "@/lib/sentry/logger";
import { fetchConfig, fetchTags } from "@/repositories";
import { getPickupArticles, getPopularArticles } from "@/services/article";
import { getCategories } from "@/services/category";
import type { PagePropsOrError } from "@/types";

type CategoriesPageProps = PagePropsOrError<CategoriesProps>;

const CategoriesPage: NextPage<CategoriesPageProps> = (props) => {
  return props.error ? <ErrorPage statusCode={props.error.statusCode} /> : <Categories {...props} />;
};

export const getStaticProps: GetStaticProps<CategoriesPageProps> = async () => {
  try {
    const [config, categories, tags, pickup, popularArticles] = await Promise.all([
      fetchConfig(),
      getCategories(),
      fetchTags(),
      getPickupArticles(new Date()),
      getPopularArticles(),
    ]);

    return {
      props: {
        categories,
        config,
        pickup,
        popularArticles,
        tags,
      },
    };
  } catch (error) {
    if (error instanceof Error) {
      await sentryLogServer(error);
    }

    return {
      props: {
        error: {
          statusCode: 500,
        },
      },
    };
  }
};

export default CategoriesPage;
