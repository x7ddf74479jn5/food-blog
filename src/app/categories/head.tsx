import { HtmlHeadBase, HtmlHeadSeo } from "@/components/meta/HtmlHead";
import { generateWebpageJsonLd, JsonLdScript } from "@/components/meta/JsonLd";
import { getCategoriesPageMeta } from "@/components/pages/categories/Categories/meta";
import { fetchConfig } from "@/repositories";

const Head = async () => {
  const [config, meta] = await Promise.all([fetchConfig(), getCategoriesPageMeta()]);

  return (
    <head>
      <HtmlHeadBase />
      <HtmlHeadSeo {...meta} />
      <JsonLdScript contents={[generateWebpageJsonLd({ config })]} />
    </head>
  );
};

export default Head;
