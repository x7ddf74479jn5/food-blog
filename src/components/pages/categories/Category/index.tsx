import { DefaultLayout } from "@/components/layouts";
import { ArticleSWRContainer } from "@/components/model/article/ArticleSWRContainer";
import { getCategoryPageMeta } from "@/components/pages/categories/Category/meta";
import { getArticles } from "@/services/article";
import { getCategory } from "@/services/category";
import { getBackLinks, urlTable } from "@/utils/paths/url";

export type CategoryProps = {
  slug: string;
};

export const Category = async ({ slug }: CategoryProps) => {
  const category = await getCategory(slug);
  const fallbackData = await getArticles({ filters: `category[equals]${category.id}`, limit: 10, offset: 0 });
  const { title: pageTitle, url } = await getCategoryPageMeta(slug);
  const backLinks = getBackLinks([urlTable.home, urlTable.categories]);
  const queryOptions = { filters: `category[equals]${category.id}` };

  return (
    <DefaultLayout pageTitle={pageTitle} url={url} backLinks={backLinks}>
      <h1 className="mb-8">カテゴリー： {category.name}</h1>
      <div className="min-h-screen w-full">
        <ArticleSWRContainer fallbackData={fallbackData} queryOptions={queryOptions} />
      </div>
    </DefaultLayout>
  );
};
