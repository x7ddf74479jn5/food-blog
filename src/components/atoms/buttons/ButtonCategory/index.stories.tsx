import { mockCategories } from "@mocks/data/categories";
import type { ComponentMeta, ComponentStory } from "@storybook/react";

import { ButtonCategory } from ".";

export default {
  title: "atoms/buttons/ButtonCategory",
  component: ButtonCategory,
} as ComponentMeta<typeof ButtonCategory>;

const Template: ComponentStory<typeof ButtonCategory> = (args) => <ButtonCategory {...args} />;

export const Default = Template.bind({});
Default.args = { category: mockCategories.rice };
