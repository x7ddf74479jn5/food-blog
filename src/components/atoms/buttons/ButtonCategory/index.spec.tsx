import { mockCategories } from "mocks/data/categories";
import renderer from "react-test-renderer";

import ButtonCategory from ".";

describe("components/atoms/buttons/ButtonCategory", () => {
  it("snapshot", () => {
    const tree = renderer.create(<ButtonCategory category={mockCategories.rice} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
