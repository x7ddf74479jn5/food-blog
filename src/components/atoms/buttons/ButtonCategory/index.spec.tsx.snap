// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/buttons/ButtonCategory snapshot 1`] = `
<a
  href="/articles/categories/rice"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="inline-block py-1 px-2 text-black dark:text-white rounded-lg border-2 border-gray-700 dark:border-gray-300"
  >
    ご飯物
  </div>
</a>
`;
